File: 백준/Silver/30804. 과일 탕후루/과일 탕후루.cc
#include <iostream>
using namespace std;
int table[200000];
int tang[10];
int main(void)
{
	ios::sync_with_stdio(0); cin.tie(0); cout.tie(0);
	int N,answer=1,f=0,temp=0;
	cin >> N;
	for (int i = 0; i < N; i++)
	{
		cin >> table[i];
		if (!tang[table[i]])temp++;
		tang[table[i]]++;
		while (temp > 2)
		{
			tang[table[f]]--;
			if (!tang[table[f]])temp--;
			f++;
		}
		answer = max(answer, i - f + 1);
	}
	cout << answer;
	return 0;
}

File: 백준/Silver/30804. 과일 탕후루/과일 탕후루.kt
import kotlin.math.max
fun main()
{
    val br=System.`in`.bufferedReader()
    val bw=System.`out`.bufferedWriter()
    var temp=0
    var answer=0
    var f=0
    var tang=IntArray(10){0}
    val N=br.readLine().toInt()
    val table=br.readLine().split(" ").map{it.toInt()}.toIntArray()
    for(i in 0..<N)
    {
        tang[table[i]]++
        if(tang[table[i]]==1)temp++
        while (temp > 2)
        {
            tang[table[f]]--
            if (tang[table[f]]==0)temp--
            f++;
        }
        answer = max(answer, i - f + 1)
    }
    bw.write("$answer")
    bw.flush()
    bw.close()
}